### DISCUSSION TITLE
I'm giving up on the configuration plist

### DISCUSSION THREAD

**Discussion by anfalas** [IBM/mac-ibm-enrollment-app#28] [General]
Hi Simone and team,   
I was very excited seeing, that IBM's enrollment app is still alive and was engineered by ground up. Your code is clean, understandable, well documented but with one major drawback: you technical documentation of the configuration plist is good, but if one is not familiar with your internal workflows and ideas it's hard to understand.    
I'm certainly not a newbie, but some more background on what the app basically does and in what order would be very helpful. 
Just a few examples on what I mean:
- why does the app need policies? Are they called directly by the app or are they available to the user by clicking them during enrollment?
- where does the userInfo coming from? Actually the user is logged in on his Mac and the information like name etc. is known. What is the background for this, how is this thought?
- what are bundles? Single Apps, Bundles of apps, MDM delivered Apps or App Store Apps triggered by MDM?
- what choices does selectedRegistrationInfo contain? Are they stored for later, maybe for the next fresh installation of the Mac?
- the same for registration - what will be registered?

Sorry, if you are deep into your concepts my questions may be trivial. And maybe an explanation of the underlying concepts will help me (and possibly others) to what you were thinking or had in mind. This and maybe a demo plist file along with a few screenshots would also help to understand how the config will be used.

At my university, we've been using Jamf Pro for a long time, but it's about time to make enrollment and onboarding more flexible and personalized. I had the pleasure of meeting Fletcher Previn in person several years ago. The concepts he presented together with the app convinced me to give it a try. Back then, your app looked different. With the old version you could guess from the code which part does what. The new flexibility of the new app makes the whole thing very exciting. And so, of course, we're dying to use your new enrollment app.    
But I'm suffering like an ox in front of the mountain, as they say in Germany: I'm not getting anywhere. So maybe you can help me (and possibly other MDM admins) out with some clarification and more documentation of the concepts and examples how to use the whole thing.

Thank you very much!
Gregor

**Comment by SMartorelli** [comment]
Hi @anfalas, sorry for the delay of this answer but I completely missed the new discussion because the web hook didn't worked properly. The new structure of the enrollment app offer more flexibility to the admin to configure it with their own workflow. The old app was a bit limited on what you can do during the "Registration Phase" (phase in which the app collect user information and save it inside the app configuration plist) and the "Bundle Installation Phase" (phase in which the app provide to the user a set of available "bundles" - collection of apps - to let them select what to install or not).

The "core" of the app didn't changed much so we always have 2 main phases that at the end trigger two different Jamf policies:
- [registrationPolicy](https://github.com/IBM/mac-ibm-enrollment-app/wiki/Configuration-file-(.plist)#jamfpoliciesstore)
- [bundleInstallationPolicy](https://github.com/IBM/mac-ibm-enrollment-app/wiki/Configuration-file-(.plist)#jamfpoliciesstore)
These are the policies trigger that you can use to run your "registration" and "bundle installation" workflow with the collected information from the app.

To answer your questions:
- The app needs 3 policies: one to trigger the registration of the user (ideally a policy that when triggered run a script to collect the registration information from the app plist. If you don't need to collect any registration information from the user skip the user registration phase), one to trigger the bundle installation (ideally a policy that when triggered run a script to collect the selectedBundles values from the app plist and run the installation of the packages for those) and one to trigger a policy to remove the framework from the user device (unenrollment - button visible on the Welcome Page). The app just run it if needed with a simple `jamf policy -event POLICY_TRIGGER`;
- In the userInfo object you just need to define the user name to let the App show it on the welcome page;
- Bundles are collection of apps, for example all the app needed for the network, productivity, communications and so on;
- selectedRegistrationInfo contains all the choices the user made during the registration phase, if to provide a registration phase and what to ask to the user in it it's your choice;
- same as above; 

We've dismissed the use of this app in our internal enrollment workflow for almost 2 years now and we adopted a more flexible and simplified approach with the combination of the [Mac@IBM Enrollment Helper](https://github.com/IBM/mac-ibm-enrollment-helper) (that help the user to download the enrollment profile and install it) and [IBM Notifier](https://github.com/IBM/mac-ibm-notifications) (a macOS agent we made to send communications to the user and so provide useful information during the enrollment process).

**Reply by anfalas** [reply]
Hi Simone,

thank you for clarifying, that helps us. I think I understand what is behind each step and what a corresponding plist might look like. I find your comment that you at IBM are now designing the enrollment process quite differently exciting.

Thanks a lot!!!

Gregor
